### PR TITLE
Blog - How to set up a Reverse Proxy in Kubernetes

### PR DESCRIPTION
**Pull request by msali123**
Draft

### CODE REVIEW

**Comment by msali123** [comment]
@balapriyac 
You can review the blog just one image is missing ill as I get from the author

[Preview Link](https://643fe19ac427180f7380fb27--ephemeral-narwhal-3ac3b2.netlify.app/blog/setup-reverse-proxy-kubernetes-nginx/)

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md [hunk 1]
```diff
@@ -0,0 +1,339 @@
+---
+title: "How to set up a Reverse Proxy in Kubernetes"
```
Comment (balapriyac) [inline blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:2]:
```suggestion
title: "How to Set Up a Reverse Proxy in Kubernetes"
```

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+internal-links:
+ - Reverse Proxy
+ - Kubernetes
+ - Performance
+ - Flask
+---
+
+Setting up a reverse proxy in kubernetes can seem a bit overwhelming if it is your first time coming across them. However, it can greatly improve the performance and reliability of your application.
+
+In this article, I will walk you through the steps to set up a reverse proxy in Kubernetes. We will cover how to configure the Nginx server, create a Kubernetes [deployment](/blog/deployment-strategies), and set up a service to expose the deployment to the outside world.
```
Comment (balapriyac) [inline blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:17]:
```suggestion
In this article, I will walk you through the steps to set up a reverse proxy in Kubernetes. We will cover how to configure the Nginx server, create a Kubernetes deployment, and set up a service to expose the deployment to the outside world.
```

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+To follow this tutorial, you will have to meet these requirements:
+
+- You have a working knowledge of Kubernetes and Docker.
+- You have [Minikube](https://minikube.sigs.k8s.io/docs/start/) installed.
+- You have [Kubectl](https://kubernetes.io/docs/tasks/tools/) installed.
+- You have [Docker](https://docs.docker.com/engine/install/) installed.
+
+## What Are Reverse Proxies
```
Comment (balapriyac) [inline blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:26]:
```suggestion
## What Are Reverse Proxies?
```

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+There are some advantages that come with using Reverse Proxies, some of which include:
+
+- Improved security
+- Increased server performance
+- Improved scalability
+- Better protocol translation, etc.
+
+## Setting Up a Reverse Proxy In Kubernetes
+
+![Setup]]({{site.images}}{{page.slug}}/setup.png)\
```
Comment (balapriyac) [inline blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:63]: @msali123 is this the image you're waiting for?

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+With the above steps, you are done with creating your flask server, creating a Docker image for your server, and pushing it to DockerHub, let's start configuring your reverse proxy.
+
+### Setting Up Your Nginx Server
+
+In this section, you will configure Nginx as a reverse proxy and build a Docker image for it.
+
+Create a separate folder called **custom_niginx**.This folder will house your Dockerfile for the Nginx image, and custom Nginx configuration files.
+
+Next, Create a new file called **nginx.conf** in the **custome_nginx folder. This file will hold the custom configurations for your reverse proxy server.
```
Comment (balapriyac) [inline blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:134]:
```suggestion
Next, Create a new file called **nginx.conf** in the **custome_nginx** folder. This file will hold the custom configurations for your reverse proxy server.
```

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+~~~{ caption="Output"}
+Forwarding from 127.0.0.1:5000 -> 5000
+Forwarding from [::1]:5000 -> 5000
+~~~
+
+This forwards traffic from port 5000 on your local machine to port 5000 on the backend-svc Kubernetes Service.
+
+Now when you visit localhost:5000 on your local machine, you should see your flask server up and running:
+
+[Flask Homepage](https://i.imgur.io/iaQtJtX_d.webp)
```
Comment (balapriyac) [inline blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:254]: @msali123 this Flask homepage should be an image. It's a markdown formatting error by the author that the editor also failed to spot. Please update.

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer. Please go through the requested changes.

File: blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md (hunk 6 above)
Comment (msali123) [reply blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:254]: Yes Yes @balapriyac I am waiting for this image.

Comment (balapriyac) [reply blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:254]: @msali123 there's one more image that's missing then? Please refer to my previous comment on setup.png.

Comment (msali123) [reply blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:254]: there was an extra bracket there 😬

Comment (msali123) [reply blog/_posts/2023-06-21-setup-reverse-proxy-kubernetes-nginx.md:254]: Changes done but still I'll push once I get the image from author

**Comment by msali123** [comment]
@balapriyac [Changes done](https://64411c0ab0a8720fabfc34d7--ephemeral-narwhal-3ac3b2.netlify.app/blog/setup-reverse-proxy-kubernetes-nginx/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**